domain-models-runtime/src/main/java/org/folio/rest/tools/utils/TenantLoading.java

```diff
@@ -181,84 +179,70 @@ private static void endWithXHeaders(HttpClientRequest req, Map<String, String> h
+        return getIdBase(url.getPath());
       case CONTENT:
         JsonObject jsonObject = new JsonObject(content);
         String id = jsonObject.getString(loadingEntry.idProperty);
         if (id == null) {
-          log.warn("Missing property "
-            + loadingEntry.idProperty + " for url=" + url.toString());
-          f.handle(Future.failedFuture("Missing property "
-            + loadingEntry.idProperty + " for url=" + url.toString()));
-          return null;
+          String msg = "Missing property " + loadingEntry.idProperty + " for url=" + url;
+          log.warn(msg);
+          return Future.failedFuture(msg);
         }
-        return StringUtil.urlEncode(id);
+        return Future.succeededFuture(StringUtil.urlEncode(id));
       case RAW_PUT:
       case RAW_POST:
         break;
     }
-    return null;
+    return Future.succeededFuture(null);
   }
 
-  private static void handleException(Throwable ex, String lead, Future<Void> f) {
-    String diag = lead  + ": " + ex.getMessage();
+  private static String handleException(Throwable ex, String lead) {
+    String diag = lead + ": " + ex.getMessage();
     log.error(diag, ex);
-    if (!f.isComplete()) {
-      f.handle(Future.failedFuture(diag));
-    }
+    return diag;
   }
 
-  private static void handleException(Throwable ex, HttpMethod method, String uri, Future<Void> f) {
-    handleException(ex, method.name() + " " + uri, f);
-  }
-
-  private static String getContent(URL url, LoadingEntry loadingEntry, Future<Void> f) {
+  static Future<String> getContent(URL url, LoadingEntry loadingEntry) {
     try {
       String content = IOUtils.toString(url, StandardCharsets.UTF_8);
       if (loadingEntry.contentFilter != null) {
-        return loadingEntry.contentFilter.apply(content);
+        return Future.succeededFuture(loadingEntry.contentFilter.apply(content));
       }
-      return content;
+      return Future.succeededFuture(content);
     } catch (IOException ex) {
-      handleException(ex, "IOException for url " + url.toString(), f);
-      return null;
+      return Future.failedFuture(handleException(ex, "IOException for url " + url.toString()));
     }
   }
 
-  private static void loadURL(Map<String, String> headers, URL url,
-    HttpClient httpClient, LoadingEntry loadingEntry, String endPointUrl,
-    Future<Void> f) {
+  private static Future<Void> loadURL(Map<String, String> headers, URL url,
+                                      WebClient httpClient, LoadingEntry loadingEntry, String endPointUrl) {
 
-    final String content = getContent(url, loadingEntry, f);
-    if (f.isComplete()) {
-      return;
-    }
-    String id = getId(loadingEntry, url, content, f);
-    if (f.isComplete()) {
-      return;
-    }
+    return getContent(url, loadingEntry)
+        .compose(content -> getId(loadingEntry, url, content)
+            .compose(id -> loadURL(headers, content, id, httpClient, loadingEntry, endPointUrl)));
```

Comment:
```suggestion
        .compose(content -> getId(loadingEntry, url, content))
        .compose(id -> loadURL(headers, content, id, httpClient, loadingEntry, endPointUrl));
```

Reply:
Can't do: content not found. Am I blind here?

Reply:
You are right, content is needed by loadURL.